BUILDING.md

```diff
@@ -0,0 +1,191 @@
+# Building instructions
+
+## Linux
+
+Dependencies:
+
+- `cmake` >= 3.1.0
+- `zlib`
+- `qt` >= 5.5 ( >= 5.6 is required if you want to rearrange tabs using
+drag&drop)
+- `Python3.5+`
+- `python3-msgpack`
+- `python3-openssl` or `python3-pyopenssl` (depending on your distro)
+- `OpenSSL` >= 1.0.0, < 1.1.0
+
+Optional dependencies: (recommended for developing)
+- `clang-tidy`
+- `clang-format`
+- `libclang-common-dev`
+
+Optional dependencies needed for running tests:
+- `gtest`
+
+If your distribution has -dev or -devel packages, you'll also need ones
+corresponding to the dependencies above.
+
+On Ubuntu it can be done like this:
+```bash
+apt-get install cmake zlib1g-dev qtbase5-dev g++ python3 python3-venv \
+    python3-dev python3-msgpack python3-openssl libffi-dev libssl-dev \
+    clang-tidy-4.0 clang-format-4.0 libclang-common-4.0-dev
+```
+
+To build:
+```bash
+$ mkdir build
+$ cd build
+$ cmake -D CMAKE_BUILD_TYPE=RelWithDebInfo ..
+$ make
+```
+
+To install [which is optional], use:
+```bash
+$ make install
+```
+
+To run Veles from build directory we recommend creating virtualenv for Veles:
+```bash
+$ python3 -m venv ~/venv/veles
+$ . ~/venv/veles/bin/activate
+(veles) $ pip install -r ../python/requirements.txt
+```
+
+Then to run Veles you will need to activate the venv before running Veles:
+```bash
+$ . ~/venv/veles/bin/activate
+(veles) $ ./veles
+```
+
+If you want to install to a non-default directory, you'll also need to pass
+it as an option to cmake before building, eg.:
+```bash
+$ cmake -D CMAKE_INSTALL_PREFIX:PATH=/usr/local ..
+```
+
+Python tests dependencies:
+
+- `git`
+- `Python2.7`
```

Comment:
unfortunately, tox currently needs exactly py3.5 to work (3.6+ won't do) -- needs to be mentioned here or fixed